src/common/hooks/home.ts

```diff
@@ -0,0 +1,84 @@
+import { useWallet, WalletContextState } from '@solana/wallet-adapter-react';
+import { useEffect } from 'react';
+import { FeaturedProfilesQuery, useFeaturedProfilesLazyQuery } from 'src/graphql/indexerTypes';
+import { ProfilePreviewData } from '../components/elements/ProfilePreviewCard';
+
+export interface QueryContext<T> {
+  data?: T | undefined;
+  loading: boolean;
+  error?: Error | undefined;
+}
+
+//TODO remove once other profiles have enough followers to preclude this one in the backend
+const DISALLOWED_PROFILES: string[] = ['ho1aVYd4TDWCi1pMqFvboPPc3J13e4LgWkWzGJpPJty'];
+
+function useFeaturedProfilesData() {
+  const wallet: WalletContextState = useWallet();
+  const [featuredProfilesQuery, queryContext] = useFeaturedProfilesLazyQuery();
+
+  useEffect(() => {
+    async function query(): Promise<void> {
+      await featuredProfilesQuery({ variables: { limit: 25, userWallet: wallet?.publicKey } });
+    }
+
+    if (!queryContext.loading && !queryContext.called) {
+      query();
+    }
+  }, [wallet, featuredProfilesQuery, queryContext.called, queryContext.loading]);
+
+  console.log(queryContext.data);
+
+  return queryContext;
+}
+
+// fetching the list of profiles to feature (only need their address)
+export function useFeaturedProfiles(): QueryContext<string[]> {
+    console.log('useFeaturedProfiles');
+  const queryContext = useFeaturedProfilesData();
+
+  //TODO implement caching of transformed object
+  return {
+    data:
+      queryContext.data?.followWallets
+        .filter((w) => !DISALLOWED_PROFILES.includes(w.address))
+        .map((w) => w.address) ?? [],
+    loading: queryContext.loading,
+    error: queryContext.error,
+  };
+}
+
+// fetching data for a single profile preview card
+export function useProfilePreview(address: string): QueryContext<ProfilePreviewData> {
+  console.log('useProfilePreview');
+    const queryContext = useFeaturedProfilesData();
+
+  const gqlObject: FeaturedProfilesQuery['followWallets'][0] | null | undefined =
+    queryContext.data?.followWallets.find((w) => w.address === address);
+
+  let error: Error | undefined = queryContext.error;
+  if (!error && gqlObject) {
+    if (gqlObject.nftCounts.created === null || gqlObject.nftCounts.created === undefined) {
+      error = new Error('Missing count of NFTs created.');
+    } else if (gqlObject.nftCounts.owned === null || gqlObject.nftCounts.owned === undefined) {
+      error = new Error('Missing count of NFTs collected.');
+    }
+  }
```

Comment:
If a fragment is used don't need to throw errors as the component says what data it needs and doesn't need to check.

Reply:
I dont think fragments fix this issue.

Any data which the backend may optionally return will be potentially nullish on the frontend and so the frontend will may still need to check nullity of required data.

In this case, the Graph QL API shows that the NFT counts are required, so it doesnt apply from the perspective of the backend.

However, the Apollo generated types make pretty much everything nullable, meaning that the frontend still doesnt have a guarantee that the data will be there. So, in fact this check is still necessary.